packages/expo-modules-core/src/requireNativeModule.web.ts

```diff
@@ -1,7 +1,17 @@
-export function requireNativeModule(moduleName: string) {
-  throw new Error(`Cannot find native module '${moduleName}'`);
+export function requireNativeModule<ModuleType = any>(moduleName: string): ModuleType {
+  const nativeModule = requireOptionalNativeModule<ModuleType>(moduleName);
+
+  if (!nativeModule) {
+    throw new Error(`Cannot find native module '${moduleName}'`);
+  }
+  return nativeModule;
 }
 
-export function requireOptionalNativeModule() {
+export function requireOptionalNativeModule<ModuleType = any>(
+  moduleName: string
+): ModuleType | null {
+  if (typeof globalThis.ExpoDomWebView === 'object' && globalThis?.expo?.modules != null) {
+    return globalThis.expo?.modules?.[moduleName] ?? null;
+  }
```

Comment:
With changes from #29870 we will be registering `globalThis.expo.modules` on Web as well. Would that be helpful or problematic for DOM components?

Reply:
DOMWebView will register `globalThis.expo` beforehand and [`registerWebGlobals` will do nothing](https://github.com/expo/expo/blob/b35859009d82590135d19e0c2187b02df1ee1c74/packages/expo-modules-core/src/web/index.web.ts#L5). that will basically to let native module calls route into native than the web implementation.

in summary, it's compatible and not override each other.

Reply:
good call, i should make this more compatible